### DISCUSSION TITLE
is socket data  ordered sequential using receiving async client?

### DISCUSSION THREAD

**Discussion by Stackerr** [miguelgrinberg/python-socketio#865] [Q&A] [answered]
hello, miguel. 
I'm using python-socketio.

I'm receive socket data using AsyncClient.
data is necessary to sequential. 
but I think async doesn't guarantee the sequential ordering at create time.

is it guarantee the sequence of data? 
if not, how can I  deal with the socket data with set ordering? 

always I really thank you about your support.

**Comment by miguelgrinberg** [comment, marked as answer]
Incoming WebSocket events are dispatched to your application in the order they are received, which is also the order in which they were sent. One possible complication is that sometimes you may end up having two or more handlers running concurrently.

For example, let's say event A is dispatched to your application, and while this event handler is running, event B arrives. With the default configuration, event B will be dispatched, even though the handler for A hasn't completed. If you want the event handlers to be serialized, set `async_workers=False` when you create your server. Then in the A and B example, B will be received, but it will wait in a queue until the handler for A returns, and only then the handler for B will be called.

**Reply by Stackerr** [reply]
I use only one handler.
you mean, I use only one handler that receive data.
receiving data using socketio.asyncClient instance guarantee data receive by create time?

i know about async is not guarantee ordering in Javascript.

**Reply by miguelgrinberg** [reply]
Each event is dispatched in the order that is read from the WebSocket connection. As I said above, if you want to ensure that your events run individually, set `async_workers=False` in your server and that will prevent any two event handlers from overlapping.
